src/batch/src/executor2/insert.rs

```diff
@@ -69,99 +70,94 @@ impl InsertExecutor {
-        }
-
-        let source_desc = self.source_manager.get_source(&self.table_id)?;
-        let source = source_desc.source.as_table_v2().expect("not table source");
-
-        let mut notifiers = Vec::new();
-
-        while let Some(child_chunk) = self.child.next().await? {
-            let len = child_chunk.cardinality();
-            assert!(child_chunk.visibility().is_none());
-
-            // add row-id column as first column
-            let mut builder = I64ArrayBuilder::new(len).unwrap();
-            for _ in 0..len {
-                builder
-                    .append(Some(source.next_row_id(self.worker_id)))
-                    .unwrap();
-            }
-
-            let rowid_column = once(Column::new(Arc::new(ArrayImpl::from(
-                builder.finish().unwrap(),
-            ))));
-            let child_columns = child_chunk.into_parts().0.into_iter();
-
-            let columns = if self.frontend_v2 {
-                // Materialize plan is assembled manually with Rust frontend, so we put the row id
-                // column to the first.
-                rowid_column.chain(child_columns).collect()
-            } else {
-                // Materialize plan is generated by the optimizer with Java frontend, so we should
-                // put row id column to the last to match the behavior of mview.
-                child_columns.chain(rowid_column).collect()
-            };
-            let chunk = StreamChunk::new(vec![Op::Insert; len], columns, None);
-
-            let notifier = source.write_chunk(chunk)?;
-            notifiers.push(notifier);
-        }
-
-        // Wait for all chunks to be taken / written.
-        let rows_inserted = try_join_all(notifiers)
-            .await
-            .map_err(|_| {
-                RwError::from(ErrorCode::InternalError(
-                    "failed to wait chunks to be written".to_owned(),
-                ))
-            })?
-            .into_iter()
-            .sum::<usize>();
-
-        // create ret value
-        {
-            let mut array_builder = PrimitiveArrayBuilder::<i64>::new(1)?;
-            array_builder.append(Some(rows_inserted as i64))?;
-
-            let array = array_builder.finish()?;
-            let ret_chunk = DataChunk::builder()
-                .columns(vec![Column::new(Arc::new(array.into()))])
-                .build();
-
-            self.executed = true;
-            Ok(Some(ret_chunk))
-        }
+    fn identity(&self) -> &str {
+        &self.identity
     }
 
-    async fn close(&mut self) -> Result<()> {
-        self.child.close().await?;
-        trace!("Cleaning insert executor.");
-        Ok(())
+    fn execute(self: Box<Self>) -> BoxedDataChunkStream {
+        self.do_execute()
     }
+}
 
-    fn schema(&self) -> &Schema {
-        &self.schema
-    }
+impl InsertExecutor2 {
+    #[try_stream(boxed, ok = DataChunk, error = RwError)]
+    async fn do_execute(mut self: Box<Self>) {
+        if !self.executed {
```

Comment:
This field is no longer necessary since `do_execute` takes the ownership of this executor.